src/metabase/search/impl.clj

```diff
@@ -232,6 +232,13 @@
                              [:and
                               [:= :bookmark.card_id :card.id]
                               [:= :bookmark.user_id (:current-user-id search-ctx)]])
+      (sql.helpers/where [:or
+                          [:= nil :card.dashboard_id]
+                          (when (:include-dashboard-questions? search-ctx)
+                            [:exists
+                             {:select 1
+                              :from [:report_dashboardcard]
+                              :where [:= :card_id :card.id]}])])
```

Comment:
I don't understand the logic here. Why not just omit the `:= nil :card.dashboard_id` check when we want to include dashboard questions?

(If there's a reason, could you make it clear in the code?)

Reply:
Good call - I added some explanatory comments. The oddity here is that if we end up with a Dashboard Question that has no DashboardCards, we want it to "disappear" from the instance. This is kind of an artifact of the weirdness of having a Dashboard Question be a separate entity from the actual link to the Dashboard itself - in an ideal world, it would be impossible to represent the situation where you have a DQ but it's not present on a dashboard. But... because it is possible to represent that situation (and it might happen, e.g. if something goes wrong during the card creation process), we want to handle it correctly and hide it from search.